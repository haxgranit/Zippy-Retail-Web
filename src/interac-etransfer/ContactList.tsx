import placeholder from './ContactList.png';

export default function ContactList() {
  return (
    <img src={placeholder} alt="placeholder" style={{ width: '100%' }} />
  );
}
