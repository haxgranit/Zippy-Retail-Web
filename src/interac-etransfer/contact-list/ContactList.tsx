import {
  Col,
  Row,
  Card,
  Table,
  Button,
} from 'react-bootstrap';
import CONTACTS from './Contacts';
import CommonHeader from '../../common/CommonHeader';

export default function ContactList() {
  return (
    <>
      <CommonHeader title="CONTACT LIST" print />
      <Row>
        <Col md={9}>
          <h4 className="mt-4">Add, Edit or Delete a Contact</h4>
          <ul className="mt-4">
            <li>
              To edit a contact&apos;s details, select the contact&apos;s name
            </li>
            <li>
              <a href="/" className="text-black">
                Edit your Interac e-Transfer alert preferences
              </a>
            </li>
            <li>
              <a href="/" className="text-black">
                Add New Contact
              </a>
            </li>
            <li>
              Note: Contacts names can not be identical. Please verify and edit
              (if necessary)
            </li>
          </ul>
        </Col>
        <Col md={3}>
          <Card className="p-3 bg-white mt-2">
            <Card.Title>You can also:</Card.Title>
            <Card.Body>
              <a
                href="/"
                className="text-decoration-none text-black"
                style={{
                  fontSize: 12,
                  marginTop: 5,
                  marginBottom: 5,
                  display: 'block',
                }}
              >
                Register for Interac e-Transfer Autodeposit
                <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                  {'>'}
                </span>
              </a>
              <a
                href="/"
                className="text-decoration-none text-black"
                style={{
                  fontSize: 12,
                  marginTop: 5,
                  marginBottom: 5,
                  display: 'block',
                }}
              >
                Make an additional loan payment
                <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                  {'>'}
                </span>
              </a>
              <a
                href="/"
                className="text-decoration-none text-black"
                style={{
                  fontSize: 12,
                  marginTop: 5,
                  marginBottom: 5,
                  display: 'block',
                }}
              >
                Order cheques
                <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                  {'>'}
                </span>
              </a>
            </Card.Body>
          </Card>
        </Col>
        <Col md={12}>
          <Table className="mt-4">
            <thead>
              <tr
                className="bg-light"
                style={{ borderTop: '1px solid #c5c5c5' }}
              >
                <th>Contact Name</th>
                <th>Notification Language</th>
                <th>Details</th>
                <th>Delete</th>
              </tr>
            </thead>
            <tbody style={{ borderTop: 'none' }}>
              {CONTACTS.map((item) => (
                <tr key={item.key}>
                  <td>{item.name}</td>
                  <td>{item.lang}</td>
                  <td>
                    {item.email && (
                      <div>{item.email}</div>
                    )}
                    {item.phone && (
                      <div>{item.phone}</div>
                    )}
                  </td>
                  <td>
                    <Button variant="light" className="text-black d-flex">
                      <div
                        style={{
                          width: 20,
                          height: 25,
                          border: '1px dotted grey',
                          textAlign: 'center',
                          marginRight: 10,
                        }}
                        title="print"
                      >
                        P
                      </div>
                      Delete
                    </Button>
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
          <div
            className="mt-4 p-3"
            style={{
              borderTop: '1px solid #e5e5e5',
              borderBottom: '1px solid #e5e5e5',
            }}
          >
            <Row className="d-flex justify-content-end">
              <Button
                variant="danger"
                className="d-flex"
                style={{ width: 'auto' }}
              >
                Add New Contact
              </Button>
            </Row>
          </div>
          <div className="mt-2">
            <div className="d-flex">
              <div
                style={{
                  width: 20,
                  height: 25,
                  border: '1px dotted grey',
                  textAlign: 'center',
                  marginRight: 10,
                }}
              >
                P
              </div>
              <span style={{ fontWeight: 'bold' }}>Note:</span>
            </div>
            <p className="d-flex align-items-end">
              <span>Your use of Interac e-Transfer is subject to the </span>
              <Row>
                <Col md={12} className="d-flex align-items-center mt-2">
                  <div
                    style={{
                      width: 20,
                      height: 25,
                      border: '1px dotted grey',
                      textAlign: 'center',
                      marginRight: 10,
                    }}
                  >
                    P
                  </div>
                  <span>
                    Interac e-Transfer Terms and Conditions (PDF, 197KB).
                  </span>
                </Col>
              </Row>
            </p>
          </div>
        </Col>
      </Row>
    </>
  );
}
