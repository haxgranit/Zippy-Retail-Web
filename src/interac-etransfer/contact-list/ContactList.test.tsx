import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ContactListContent } from './ContactList';

const ReactTestRenderer = require('react-test-renderer');

describe('Contact List Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <ContactListContent
          contacts={[]}
          onDeletePressed={jest.fn()}
        />

      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render contact list', () => {
    const { getByText } = render(
      <BrowserRouter>
        <ContactListContent
          contacts={[]}
          onDeletePressed={jest.fn()}
        />

      </BrowserRouter>,
    );
    expect(getByText('Add, Edit or Delete a Contact')).toBeInTheDocument();
  });
});
