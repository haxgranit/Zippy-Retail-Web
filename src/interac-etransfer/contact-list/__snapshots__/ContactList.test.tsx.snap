// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact List Component matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h4
      className="mt-4"
    >
      Add, Edit or Delete a Contact
    </h4>
    <ul
      className="mt-4"
    >
      <li>
        To edit a contact's details, select the contact's name
      </li>
      <li>
        <a
          className="text-black"
          href="/"
        >
          Edit your Interac e-Transfer alert preferences
        </a>
      </li>
      <li>
        <a
          href="/add/"
          onClick={[Function]}
        >
          <button
            className="text-black btn btn-link"
            disabled={false}
            type="button"
          >
            Add New Contact
          </button>
        </a>
      </li>
      <li>
        Note: Contacts names can not be identical. Please verify and edit (if necessary)
      </li>
    </ul>
  </div>
  <div
    className="col-md-3"
  >
    <div
      className="p-3 bg-white mt-2 card"
    >
      <div
        className="card-title h5"
      >
        You can also:
      </div>
      <div
        className="card-body"
      >
        <a
          className="text-decoration-none text-black"
          href="/"
          style={
            Object {
              "display": "block",
              "fontSize": 12,
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
        >
          Register for Interac e-Transfer Autodeposit
          <span
            style={
              Object {
                "color": "rgb(220, 53, 69)",
                "marginLeft": 5,
              }
            }
          >
            &gt;
          </span>
        </a>
        <a
          className="text-decoration-none text-black"
          href="/"
          style={
            Object {
              "display": "block",
              "fontSize": 12,
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
        >
          Make an additional loan payment
          <span
            style={
              Object {
                "color": "rgb(220, 53, 69)",
                "marginLeft": 5,
              }
            }
          >
            &gt;
          </span>
        </a>
        <a
          className="text-decoration-none text-black"
          href="/"
          style={
            Object {
              "display": "block",
              "fontSize": 12,
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
        >
          Order cheques
          <span
            style={
              Object {
                "color": "rgb(220, 53, 69)",
                "marginLeft": 5,
              }
            }
          >
            &gt;
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    className="col-md-12"
  >
    <table
      className="mt-4 table"
    >
      <thead>
        <tr
          className="bg-light"
          style={
            Object {
              "borderTop": "1px solid #c5c5c5",
            }
          }
        >
          <th>
            Contact Name
          </th>
          <th>
            Notification Language
          </th>
          <th>
            Details
          </th>
          <th>
            Delete
          </th>
        </tr>
      </thead>
      <tbody
        style={
          Object {
            "borderTop": "none",
          }
        }
      />
    </table>
    <div
      className="mt-4 p-3"
      style={
        Object {
          "borderBottom": "1px solid #e5e5e5",
          "borderTop": "1px solid #e5e5e5",
        }
      }
    >
      <div
        className="d-flex justify-content-end row"
      >
        <a
          href="/add/"
          onClick={[Function]}
        >
          <button
            className="d-flex btn btn-danger"
            disabled={false}
            style={
              Object {
                "width": "auto",
              }
            }
            type="button"
          >
            Add New Contact
          </button>
        </a>
      </div>
    </div>
    <div
      className="mt-2"
    >
      <div
        className="d-flex"
      >
        <div
          style={
            Object {
              "border": "1px dotted grey",
              "height": 25,
              "marginRight": 10,
              "textAlign": "center",
              "width": 20,
            }
          }
        >
          P
        </div>
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Note:
        </span>
      </div>
      <div
        className="d-flex align-items-end"
      >
        <span>
          Your use of Interac e-Transfer is subject to the 
        </span>
        <div
          className="row"
        >
          <div
            className="d-flex align-items-center mt-2 col-md-12"
          >
            <div
              style={
                Object {
                  "border": "1px dotted grey",
                  "height": 25,
                  "marginRight": 10,
                  "textAlign": "center",
                  "width": 20,
                }
              }
            >
              P
            </div>
            <span>
              Interac e-Transfer Terms and Conditions (PDF, 197KB).
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
